### PR TITLE
`cyclic_generator` for groups, `is_faithful` for characters; make `findfirst(fun, t::GAPGroupCharacterTable)` work

### PR DESCRIPTION
**Pull request by ThomasBreuer**
(suggested by @fieker )

- `cyclic_generator` for groups
- `is_faithful` for characters
- make `findfirst(fun, t::GAPGroupCharacterTable)` work

### CODE REVIEW

**Review by fieker — APPROVED** [review]

File: src/Groups/group_constructors.jl [hunk 1]
```diff
@@ -127,6 +127,27 @@ i.e., if `G` can be generated by one element.
...
+Group([ (1,2,3), (4,5) ])
+
+julia> cyclic_generator(g)
+(1,2,3)(4,5)
+```
+"""
+function cyclic_generator(G::GAPGroup)
+  @req is_cyclic(G) "the group is not cyclic"
+  length(gens(G)) == 1 && return gen(G,1)
+  return group_element(G, GAPWrap.MinimalGeneratingSet(G.X)[1])
```
Comment (fingolfin) [inline src/Groups/group_constructors.jl:149]: Can we also get a `minimal_generating_set` function (not in this PR, but in a follow-up perhaps)?

**Review by fingolfin — APPROVED** [review]

File: src/Groups/group_constructors.jl (hunk 1 above)
Comment (ThomasBreuer) [reply src/Groups/group_constructors.jl:149]: Is GAP's `MinimalGeneraingSet` the right thing to ask for? Often it is sufficient to have a generating set of minimal length.

**Comment by thofma** [comment]
Would another option be `generator(G)` instead of `cyclic_generator(G)`? I never heard of a cyclic generator before.

**Comment by fieker** [comment]
On Fri, Apr 21, 2023 at 05:11:56AM -0700, Tommy Hofmann wrote:
> Would another option be `generator(G)` instead of `cyclic_generator(G)`? I never heard of a cyclic generator before.
So far, generator usually refers to the natural one that is used: 
C2 x C3 would have so far 2 gens of order 2 and 3. The new function will
produce some element of order 6

**Comment by fieker** [comment]
On Fri, Apr 21, 2023 at 05:11:56AM -0700, Tommy Hofmann wrote:
> Would another option be `generator(G)` instead of `cyclic_generator(G)`? I never heard of a cyclic generator before.
> 
is_cyclic_with_data...

**Comment by thofma** [comment]
Yes, `generator(G)` means give me one generator if it exists. It is not related to `generators(G)`.

**Comment by fingolfin** [comment]
Do you mean `gen(G)` versus `gens(G)` (versus `gen(G, i)`)? Because right now there are no functions `generators` / `generator` that work in general (well, there is `generators` for  `AbstractLat`... but that's all?)

**Comment by fingolfin** [comment]
What I like about `cyclic_generator` is that it is kind of obvious what it does. While `gen(G)` can be a typo for `gens(G)`. I mean, it logically "makes sense" and seems cute, but perhaps it is *too* cute...

**Comment by ThomasBreuer** [comment]
Would `generator` then be something that is interesting in other contexts as well?
For field extensions, `generator` would return a primitive element.
For univariate polynomial rings, `generator` would return the indeterminate.

**Comment by fingolfin** [comment]
That said, I don't mind much if `cyclic_generator` is renamed to `gen` *shrug*

**Comment by thofma** [comment]
I my mind the functions `gen` and `gens` are already changed to `generator` and `generators` (the name they should have).

OK, maybe `cyclic_generator` is the best we can do.

**Comment by fingolfin** [comment]
Actually, will `cyclic_generator` work for trivial groups?

**Comment by thofma** [comment]
> Actually, will `cyclic_generator` work for trivial groups?

Hm, I don't see a reason why not. Can you elaborate?

**Comment by ThomasBreuer** [comment]
> Actually, will cyclic_generator work for trivial groups?

Ah, I will update the pull request.
Thanks for the hint.
(And we have found a bug in GAP's grpconst package.)

**Pull request merged**